src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -395,6 +395,18 @@ public void CreateAccount(Address address, in UInt256 balance, in UInt256 nonce)
             PushNew(address, account);
         }
 
+        public void AddToBalanceAndCreateIfNotExists(Address address, in UInt256 balance, IReleaseSpec spec)
```

Comment:
nice, common pattern. 

WithdrawalProcessor.cs will like this method as well :)  https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Consensus/Withdrawals/WithdrawalProcessor.cs#L41